README.md

```diff
@@ -1,32 +1,179 @@
+`null` value is the default value for all reference types (objects and `String`
+data types). This means, in memory, if we do not initialize the reference type
+when we first declare it, the variable in the stack will point to `null` in the
+heap. Let's see what happens when we don't initialize a reference
+variable!
 
-- [Resource Link 1](example.com)
-- [Resource Link 2](example.com)
+## Instance and Static Variable Initialization
+
+```java
+public class Person {
+
+    private String name;
+
+    public static void main(String[] args) {
+        Person p1 = new Person();
+        System.out.println(p1.name);
+    }
+}
+```
+
+Consider the instance variable `name`.  Since `name` is not assigned
+an explicit value, it is assigned the default value `null`.
+
+![null value](https://curriculum-content.s3.amazonaws.com/6676/java-multipleclasses/null.png
+)
+
+The output of the above will look like this:
+
+```text
+null
+```
+
+## Initialization of local variables
+
+Instance variables are initialized
+to a default value; however, local variables are not.
+If we update the `main` method to declare a local
+variable `favoriteColor`, the method won't compile
+because the variable has not been assigned a value.
+
+```java
+public class Person {
+
+    private String name;  //initialized to null
+
+    public static void main(String[] args) {
+        Person p1 = new Person();
+        System.out.println(p1.name);
+
+        String favoriteColor ;    //uninitialized
+        System.out.println(favoriteColor);  //compiler error at this line
+    }
+}
+```
+
+To compile, we must assign the local variable a value.
+If we want, we can  assign `null` like so:
+
+```java
+public class Person {
+
+    private String name;  //initialized to null
+
+    public static void main(String[] args) {
+        Person p1 = new Person();
+        System.out.println(p1.name);
+
+        String favoriteColor = null ;    //initialized to null
+        System.out.println(favoriteColor);  //no error 
+    }
+}
+```
+
+The output of the above will look like this:
+
+```text
+null
+null
+```
+
+Notice that the `null` is in all lowercase letters. The casing of the word
+matters. If the code were to be written `String favoriteColor = NULL` it would not
+compile.
+
+The `null` value is only to be used with reference types and not primitives. If
+we were to assign a primitive type to `null`, we would get an error about the
+type being incompatible. However, we will see in a subsequent lesson
+on type casting that we
+can use the `null` value with the wrapper classes.
+
+```java
+int x = null;    // Will result in an error at compilation
+
+Integer y = null;    // this is okay since Integer is a reference type
+```
+
+## NullPointerException
+
+The `NullPointerException` is a runtime exception that we may see occur in our
+programs from time to time. This is a more common exception to be on the lookout
+for! `NullPointerException` is thrown when we try to use an object that has a
```

Comment:
Perhaps this could be clarified a little. 
Maybe instead of saying 'when we try to *use* an object that has a value of null', you could say 'when we try to access a method or property of a variable that has a value of null'

Reply:
fixed